### PR TITLE
Introduce FileSlice

### PR DESCRIPTION
**Pull request by wiktorn**
New class providing "windowed" view over the file objects. It gives the
possibility to have a file-alike object that returns only some part of the file.
Useful for multipart upload.

### CODE REVIEW

File: src/onedrivesdk/extensions/fragment_upload.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
+# -*- coding: utf-8 -*-
+'''
+The MIT License (MIT)
+
+Copyright (c) 2015 Wiktor Niesiobędzki
```
Comment (cdmayer) [inline src/onedrivesdk/extensions/fragment_upload.py]: Actually, in the CLA it says you can't keep the copyright. Please copy/paste the license out of one of the other files as-is.

**Comment by cdmayer** [comment]
Once you fix the licenses this should be good to go!

**Comment by wiktorn** [comment]
Please wait for #11, before commiting this, as for now - this code requires changes in #11, as it calls create_session() without arguments.

**Comment by wiktorn** [comment]
Ok, so now it's independent of #11. Though I'm not sure, if my intensive rebasing will not anyway pollute the commit history

**Comment by cdmayer** [comment]
I appreciate it nonetheless. This one is approved.

**Comment by dmrickert** [comment]
+1 for getting this into master

**Pull request merged**